endpoints/endpoints.py

```diff
@@ -0,0 +1,1426 @@
+
+    Args:
+        obj: A variable of potentially many types to convert into a JSON string
+
+    Globals:
+        None
+
+    Returns:
+        str: A formatted string containing the JSON representation of obj
+    """
+    return json.dumps(obj, indent = 4, separators=(',', ': '), sort_keys = True, default = not_json_serializable)
+
+def expand_id_range(id_range):
+    """
+    Helper function to expand an ID range
+
+    Args:
+        id_range (str): An ID range in one of the following forms [ "A-E", "F" ]
+
+    Globals:
+        None
+
+    Returns:
+        expanded_ids (list): A list containing the broken out IDs in integer format such as [ A, B, C, D, E ] or [ F ]
+    """
+    expanded_ids = []
+
+    split_id_range = id_range.split("-")
+    if len(split_id_range) == 1:
+        expanded_ids.append(int(split_id_range[0]))
+    else:
+        split_id_range = list(map(int, split_id_range))
+        if split_id_range[0] >= 0 and split_id_range[0] < split_id_range[1]:
+            for subid in range(split_id_range[0], split_id_range[1]+1):
+                expanded_ids.append(subid)
+        else:
+            raise ValueError("Invalid id range: %s" % (id_range))
+
+    return expanded_ids
+
+def expand_ids(ids):
+    """
+    Take a IDs specification and expand it
+
+    Args:
+        ids (str, int, or list): A complex variable containing one of many forms of ID representation
+                                 such as [ "A-E", "A-E+H-K+M", "A", A, [ A, "C-E", "H" ] ]
+
+    Globals:
+        None
+
+    Returns:
+        new_ids (list): A list containing the broken out IDs in integer format such as [ A, B, C, D, E] or
+                        [ A, B, C, D, E, H, J, K, M ] or [ A ] or [ A ] or [ A, C, D, E, H ]
+    """
+    new_ids = []
+
+    if isinstance(ids, str):
+        subids = ids.split("+")
+        for subid in subids:
+            new_ids.extend(expand_id_range(subid))
+    elif isinstance(ids, int):
+        new_ids.append(ids)
+    elif isinstance(ids, list):
+        for id in ids:
+            if isinstance(id, int):
+                new_ids.append(id)
+            elif isinstance(id, str):
+                new_ids.extend(expand_id_range(id))
+
+    new_ids.sort()
+
+    return new_ids
+
+def is_ip(ip_address):
+    """
+    Determine if the provided variable contains an IPv4 of IPv6 address
+
+    Args:
+        ip_address (str): A string to check if it is a valid IPv4 or IPv5 address (it could be a hostname)
+
+    Globals:
+        None
+
+    Returns:
+        True or False
+    """
+    log.info("Checking to see if '%s' is an IP address" % (ip_address))
+
+    # check for IPv4
+    m = re.search(r"[1-9][0-9]{0,2}\.[1-9][0-9]{0,2}\.[1-9][0-9]{0,2}\.[1-9][0-9]{0,2}", ip_address)
+    if m:
+        return True
+
+    # check for IPv6
+    m = re.search(r"[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]{1,4}:[0-9a-fA-F]", ip_address)
```

Comment:
If we change here or in the schema, we're out of sync... not sure how to solve this though